File: compression/compress.go
package compress

import "fmt"

func compress(uncompressed string) string {
	var buf string

	if len(uncompressed) == 0 {
		return buf
	}

	letter := string(uncompressed[0])
	count := 1
	for _, next := range uncompressed[1:] {
		if letter == string(next) {
			count++
		} else {
			buf += fmt.Sprintf("%s%d", letter, count)
			count = 1
		}
		letter = string(next)
	}
	buf += fmt.Sprintf("%s%d", letter, count)

	return buf
}

File: compression/compress.py



def compress(uncompressed):
    count = 1
    compressed = ""
    if not uncompressed:
        return compressed
    letter = uncompressed[0]
    for nx in uncompressed[1:]:
        if letter == nx:
            count = count + 1
        else:
            compressed += "{}{}".format(letter, count)
            count = 1
        letter = nx
    compressed += "{}{}".format(letter, count)
    return compressed
            
if __name__ == "__main__":
    print(compress("aaabbbccccd"))